File: WEEK 1/JavaScript/Summer Job Income Calculation.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function getUserInput(prompt) {
  return new Promise((resolve) => {
    rl.question(prompt, (input) => {
      resolve(parseFloat(input));
    });
  });
}

async function main() {
  try {
    const payRatePerHour = await getUserInput('Enter your pay rate per hour (in ZAR): ');
    const hoursWorkedPerWeek = await getUserInput('Enter the number of hours worked each week: ');
    
    if (payRatePerHour < 0 || hoursWorkedPerWeek < 0) {
      console.log('Please enter non-negative values for pay rate and hours worked.');
      return;
    }

    const weeksWorked = 8; // Assuming a typical summer job lasts for 8 weeks
    const income = payRatePerHour * hoursWorkedPerWeek * weeksWorked;

    const taxRate = 0.15; // 15% tax rate
    const taxAmount = income * taxRate;

    const clothesAndSuppliesCost = 500; // Estimated cost of clothes and school supplies
    const savingsBondPurchase = 1000; // Amount spent on savings bonds

    const totalExpenses = clothesAndSuppliesCost + savingsBondPurchase;
    const remainingIncome = income - taxAmount - totalExpenses;

    console.log(`Income from summer job: ZAR ${income.toFixed(2)}`);
    console.log(`Tax amount: ZAR ${taxAmount.toFixed(2)}`);
    console.log(`Expenses on clothes and school supplies: ZAR ${clothesAndSuppliesCost.toFixed(2)}`);
    console.log(`Savings bond purchase: ZAR ${savingsBondPurchase.toFixed(2)}`);
    console.log(`Remaining income: ZAR ${remainingIncome.toFixed(2)}`);
  } catch (error) {
    console.error(error);
  } finally {
    rl.close();
  }
}

main();

File: WEEK 1/java/Summer Job Income Calculation.java
import java.util.Scanner;

public class SummerJobIncomeCalculator {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        try {
            // Get user input
            System.out.print("Enter your pay rate per hour (in ZAR): ");
            double payRatePerHour = input.nextDouble();

            System.out.print("Enter the number of hours worked each week: ");
            double hoursWorkedPerWeek = input.nextDouble();

            if (payRatePerHour < 0 || hoursWorkedPerWeek < 0) {
                System.out.println("Please enter non-negative values for pay rate and hours worked.");
                return;
            }

            int weeksWorked = 8; // Assuming a typical summer job lasts for 8 weeks
            double income = payRatePerHour * hoursWorkedPerWeek * weeksWorked;

            double taxRate = 0.15; // 15% tax rate
            double taxAmount = income * taxRate;

            double clothesAndSuppliesCost = 500; // Estimated cost of clothes and school supplies
            double savingsBondPurchase = 1000; // Amount spent on savings bonds

            double totalExpenses = clothesAndSuppliesCost + savingsBondPurchase;
            double remainingIncome = income - taxAmount - totalExpenses;

            // Output results
            System.out.println("Income from summer job: ZAR " + String.format("%.2f", income));
            System.out.println("Tax amount: ZAR " + String.format("%.2f", taxAmount));
            System.out.println("Expenses on clothes and school supplies: ZAR " + String.format("%.2f", clothesAndSuppliesCost));
            System.out.println("Savings bond purchase: ZAR " + String.format("%.2f", savingsBondPurchase));
            System.out.println("Remaining income: ZAR " + String.format("%.2f", remainingIncome));
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        } finally {
            input.close();
        }
    }
}
